### PR TITLE
DS-520 new 'Display' typographic component

### PR DESCRIPTION
**Pull request by jeremy-davis-sonarsource**

### CODE REVIEW

File: src/components/typography/Display.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { PropsWithChildren } from 'react';
+
+interface Props {
+  as?: 'p' | 'span' | 'div';
```
Comment (gregaubert) [inline src/components/typography/Display.tsx:25]: `h*` too ?

File: src/components/typography/Display.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  hasMarginBottom?: boolean;
```
Comment (gregaubert) [inline src/components/typography/Display.tsx:26]: Should we really have this property? When do we want to have margin-bottom for a display? It's not a heading right?

File: stories/typography/Display-stories.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+/* eslint-disable no-console */
```
Comment (gregaubert) [inline stories/typography/Display-stories.tsx:21]: We might not need this line anymore?

**Review by david-cho-lerat-sonarsource — APPROVED** [review]
LGTM :+1:

File: .storybook/preview.tsx [hunk 4]
```diff
@@ -32,6 +32,10 @@ const globalStyles = css`
     color: var(--echoes-color-text-default);
     background-color: var(--echoes-color-background-default);
   }
+
+  p {
+    margin: 0;
+  }
```
Comment (gregaubert) [inline .storybook/preview.tsx:38]: I think we shouldn't have that. If we expect the Display component to reset paragraphs margins then it should do so in it's implem, we can't expect the products to have this reset.

File: src/components/typography/Display.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { PropsWithChildren } from 'react';
+
+interface Props {
```
Comment (gregaubert) [inline src/components/typography/Display.tsx:24]: We are missing the className prop

File: src/components/typography/Display.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  as?: 'p' | 'span' | 'div';
+  hasMarginBottom?: boolean;
+}
+
+export function Display({
```
Comment (gregaubert) [inline src/components/typography/Display.tsx:29]: We should forward ref and rest of the props.

File: src/components/typography/Display.tsx (hunk 2 above)
Comment (jeremy-davis-sonarsource) [reply src/components/typography/Display.tsx:26]: Good point, I'll ping Marcio

File: src/components/typography/Display.tsx (hunk 1 above)
Comment (jeremy-davis-sonarsource) [reply src/components/typography/Display.tsx:25]: Specifically not, I believe

File: src/components/typography/Display.tsx (hunk 2 above)
Comment (gregaubert) [reply src/components/typography/Display.tsx:26]: I also added a comment in figma about it

File: .storybook/preview.tsx (hunk 4 above)
Comment (jeremy-davis-sonarsource) [reply .storybook/preview.tsx:38]: As discussed, it doesn't handle margins at all

**Pull request merged**